modules/rational-completion.el

```diff
@@ -76,5 +78,57 @@ folder, otherwise delete a word"
 ;; Use Embark to show bindings in a key prefix with `C-h`
 (setq prefix-help-command #'embark-prefix-help-command)
 
+;;;; Corfu
+
+;; Setup corfu for popup like completion
+(require 'corfu)
+
+(defcustom rational-code-completion-ycm-style nil
+  "Make corfu act closer to YouCompleteMe of Vim fame."
+  :type 'boolean)
+
+(when rational-code-completion-ycm-style
```

Comment:
It would be helpful to have this as a `:set` on the `defcustom` so that it is effective when set by a user. Otherwise we end up in an order-dependency problem.

Reply:
Makes sense, just not sure how that works.

Reply:
The easiest way to do this (and probably the best way for maintenance in the future) would be to define a pair of functions which handle the "ycm-style" configuration:

`(defun rational-completion-enable-ycm-style () ...`

and 

`(defun rational-completion-disable-ycm-style () ...`

You could then update the `defcustom` to support calling the appropriate function when changing the variable like this:

```
(defcustom rational-completion-ycm-style nil
  "Make corfu act closer to YouCompleteMe of Vim fame."
  :type 'boolean
  :group `rational
  :set (lambda (sym val)
         (let ((prev-val (if (boundp sym)
                             sym
                           nil)))
           (set-default sym val)
           (if val
               (rational-completion-enable-ycm-style)
             (rational-completion-disable-ycm-style)))))
```